seccomp/src/lib.rs

```diff
@@ -142,7 +142,7 @@
 //!     }
 //! }
 //!
-//! fn gen_rules() -> Vec<SyscallRuleSetWithPriority> {
+//! fn gen_rules() -> Vec<SyscallRuleSet> {
```

Comment:
* In the first commit of this PR the `Vec<(i64, Vec<SeccompRule>)>` is replaced  with newly added type `SyscallRuleSetWithPriority`. In this commit the `SyscallRuleSetWithPriority` is replaced with `SyscallRuleSet` and priority is removed. It is rather confusing this way. What is the reason behind this? Did you want to do it incrementally to make it easier to understand the changes or is it a miss?
* Upon reading the commit message I was under the impression that there was a connection between allow_syscall_if and the removal of the priority. However, after reviewing there seems to be none, right?

Reply:
1. I removed the priority after a brief review session with @dhrgit during which we came to the conclusion that we weren't using priorities at all, so this commit went on top of the initial "solution". I tried to squash the 2 commits you identified that alter the same data structure, but there were too many unrelated changes to sweep through and I dropped the effort.
1. There is no connection. During the same review session we agreed to add `allow_syscall_if` to improve readability as there was a mix of `allow_syscall` and explicitly built tuples.